### PR TITLE
Updates to Servo class and fix ESP32 servo issue on beta 55

### PR DESCRIPTION
**Pull request by brentru**
* Automatically writes min pulse width value to servo object upon creation
  * This removes 1x protobuf data transfer, reduces bandwidth 
  * @lorennorman  - Please remove the default servo write on staging, let me know here when ready to test.
* Adds  `getServoIdx` for performing index lookup
* Initialize servo struct with default values instead of init'ing within the constructor
* Fixes servo write on ESP32x platforms by using the LEDC timer width from beta 54, instead of the 16bit width from beta 55.

### CODE REVIEW

**Comment by brentru** [comment]
Artifacts are ready for testing + checks passed

**Review by lorennorman — APPROVED** [review]
Only necessary change is the misleading error message, the rest is me having fun.

File: src/components/servo/ws_servo.cpp [hunk 1]
```diff
@@ -20,12 +20,7 @@
     @brief  Constructor
 */
 /**************************************************************************/
-ws_servo::ws_servo() {
-  for (int i = 0; i < MAX_SERVO_NUM; i++) {
-    _servos[i].pin = 255;
-    _servos[i].servoObj = nullptr;
-  }
-}
+ws_servo::ws_servo() {}
```
[the remark this answers is not in the document]
Comment (brentru) [reply src/components/servo/ws_servo.cpp:23]: I'm not sure. Will read up and get back to you.

[the remark this answers is not in the document]
Comment (brentru) [reply src/components/servo/ws_servo.cpp:23]: Switching to [the default constructor](https://learn.microsoft.com/en-us/cpp/cpp/constructors-cpp?view=msvc-170#default_constructors) on the next commit.

File: src/components/servo/ws_servo.cpp [hunk 2]
```diff
@@ -92,23 +105,29 @@ bool ws_servo::servo_attach(int pin, int minPulseWidth, int maxPulseWidth,
...
-      _servos[i].servoObj->detach();
-      return;
-    }
+  int servoIdx = getServoIdx(pin);
+  if (servoIdx == -1) {
+    WS_DEBUG_PRINTLN("ERROR: Can not detach servo, not found!");
+    return;
   }
+  // default value for unallocated servo pin
+  _servos[servoIdx].pin = 255;
```
[the remark this answers is not in the document]
Comment (brentru) [reply src/components/servo/ws_servo.cpp:126]: Switch to a defined constant

File: src/components/servo/ws_servo.cpp [hunk 3]
```diff
@@ -40,6 +35,24 @@ ws_servo::~ws_servo() {
...
+*/
+/**************************************************************************/
+int ws_servo::getServoIdx(uint8_t servoPin) {
+  int servoIdx = -1;
+  for (int i = 0; i < MAX_SERVO_NUM; i++) {
+    if (_servos[i].pin == servoPin)
+      return i;
+  }
+  return servoIdx;
+}
```
[the remark this answers is not in the document]
Comment (brentru) [reply src/components/servo/ws_servo.cpp]: Thank you for the thoughts here and on slack, I've refactored around this.

**Review by lorennorman — APPROVED** [review]
💯

**Comment by lorennorman** [comment]
@brentru ok this is live on staging for you to test!

**Comment by brentru** [comment]
@lorennorman  Tested on hardware, OK'ing deploy to production

---


Prod
```
16:27:33.640 > ATTACHED servo w/minPulseWidth: 500 uS and maxPulseWidth: 500uS on pin: 27
16:27:33.648 > -> Servo Attach Response...Published!
16:27:33.917 > * NEW MESSAGE [Topic: Servo]: 
16:27:33.918 > 10 bytes.
16:27:33.918 > Decoding Servo Message...
16:27:33.924 > GOT: Servo Write
16:27:33.924 > Writing pulse width of 500uS to servo on pin#: 27
16:27:33.924 > 27
16:27:34.283 > PING!
```

Staging
```
16:32:11.316 > * NEW MESSAGE [Topic: Servo]: 
16:32:11.321 > 15 bytes.
16:32:11.321 > Decoding Servo Message...
16:32:11.327 > GOT: Servo Attach
16:32:11.327 > 27
16:32:11.327 > ATTACHED servo w/minPulseWidth: 500 uS and maxPulseWidth: 500uS on pin: 27
16:32:11.332 > -> Servo Attach Response...Published!
```

**Comment by lorennorman** [comment]
@brentru I am wondering if this change has caused "sync" to not do the right thing now, can you test that easily?

To be clear: if there is already a value on the Servo's feed, the device should move to that position on sync (right?) I'm worried I broke this and need a further tweak.

**Comment by brentru** [comment]
Does not sync on staging

**Pull request merged**